.eslintrc.js

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable import/no-commonjs */
+const defaultRules = {
+  'no-continue': 'off',
+  'valid-jsdoc': 'off',
+  'require-await': 'off',
+  'import/extensions': [
+    'error',
+    {
+      ignorePackages: true,
+      pattern: {
+        js: 'always',
+        ts: 'never',
+      },
+    },
+  ],
+};
+module.exports = {
+  extends: ['algolia', 'algolia/jest'],
+  rules: defaultRules,
+  overrides: [
+    {
+      files: ['**/*.ts'],
+      extends: ['algolia', 'algolia/jest', 'algolia/typescript'],
+      parser: '@typescript-eslint/parser',
+      parserOptions: {
+        sourceType: 'module',
+        project: './tsconfig.json',
+      },
+      rules: {
+        ...defaultRules,
```

Comment:
I don't think it's necessary to reapply rules, overrides only override what is specified

Reply:
ah yes I missed that, indeed not needed. Will fix

Reply:
Are you sure about that? That was the issue I was having, the `overrides` was overriding the whole `rules` from the top level, that's why I had to move from `.json` to `.js`

Reply:
no, it was because of the `extends`, 4df2833 is more "correct"

Reply:
No, you're right, that works. I thought I had an issue with not being able to merge an override rule with the main one, but that might have been caused by something else. Current config is lean and clean